### PR TITLE
Update for saving image to GCS

### PR DESCRIPTION
**Pull request by thompsond**

### CODE REVIEW

File: src/main/java/com/google/step/MapImage.java [hunk 1]
```diff
@@ -60,12 +60,12 @@
   private int year;
 
   /** Exact time snapshot was took. */
-  private int timeStamp;
+  private long timeStamp;
```
Comment (cgregori) [inline src/main/java/com/google/step/MapImage.java:63]: I feel like a TimeStamp object might be the best route forward here (& in the subsequent spots in the code). That way it matches up with Datastore's TimeStamp value.

File: src/main/java/com/google/step/SaveImages.java [hunk 2]
```diff
@@ -68,16 +68,15 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
-        city = city.replaceAll(" ", "_");
-        String bucketName = String.format("%s.appspot.com", PROJECT_ID);
+    private void saveImageToCloudStorage(byte[] imageData, MapImage mapImage) throws StorageException {
         Storage storage = StorageOptions.newBuilder().setProjectId(PROJECT_ID).build().getService();
-        String objectID = String.format("%s_%dx_%s_%s.png", city, zoomLevel, month, year);
-        BlobId blobId = BlobId.of(bucketName, objectID);
+        mapImage.setMonth(Integer.parseInt(month));
+        mapImage.setYear(Integer.parseInt(year));
+        mapImage.setObjectID();
+        mapImage.setTimeStamp(System.currentTimeMillis());
```
Comment (cgregori) [inline src/main/java/com/google/step/SaveImages.java:78]: Possibly TimeStampObject.setTime(System.currentTimeMillis()); instead. The oracle docs have more info

File: src/main/java/com/google/step/SaveImages.java [hunk 3]
```diff
@@ -68,16 +70,17 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
 
-    private void saveImageToCloudStorage(byte[] imageData, String city, int zoomLevel) throws StorageException {
-        // Add underscores to city names with spaces
-        city = city.replaceAll(" ", "_");
-        String bucketName = String.format("%s.appspot.com", PROJECT_ID);
+    private void saveImageToCloudStorage(byte[] imageData, MapImage mapImage) throws StorageException {
         Storage storage = StorageOptions.newBuilder().setProjectId(PROJECT_ID).build().getService();
-        String objectID = String.format("%s_%dx_%s_%s.png", city, zoomLevel, month, year);
-        BlobId blobId = BlobId.of(bucketName, objectID);
+        mapImage.setMonth(Integer.parseInt(month));
```
Comment (caprichard) [inline src/main/java/com/google/step/SaveImages.java:77]: I may be misreading the code here, but isn't month already of type int?  Why do you need to do Integer.parseInt()?

Comment (thompsond) [reply src/main/java/com/google/step/SaveImages.java:77]: The local variable is a string

Comment (caprichard) [reply src/main/java/com/google/step/SaveImages.java:77]: Ah - sure enough.  The review UI collapsed the code above this, and I ended up looking at the MapImage definition.

**Pull request merged**